pkg/util/test/util.go

```diff
@@ -576,6 +593,137 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
+				f := func() (interface{}, bool, error) {
+					logrus.Debugf("NodeWiper pod [%s] ; State: [%s]", p.Name, p.Status.Phase)
+					if !coreops.Instance().IsPodRunning(p) {
+						return nil, true, fmt.Errorf("NodeWiper pod [%s] is not running yet", p.Name)
+					}
+					return nil, false, nil
+				}
+				if _, err := task.DoRetryWithTimeout(f, timeout, interval/2); err != nil {
+					logrus.Errorf("NodeWiper pod [%s] timed out to be ready", p.Name)
+					podsNotReady = append(podsNotReady, p.Name)
+				} else {
+					logrus.Infof("NodeWiper pod [%s] is Running!", p.Name)
+					podsReady = append(podsReady, p.Name)
+				}
+			}(pod)
+		}
+		wg.Wait()
+
+		if len(podsReady) != len(expectedPxNodeNameList) {
+			return fmt.Errorf("failed to get following NodeWiper pods ready on storageNodes: [%s]", podsNotReady)
+		}
+		logrus.Debugf("All NodeWiper pods are ready on storageNodes: [%s]", podsReady)
+	}
+
+	// Validate images are correct
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get operator version, Err: %v", err)
+	}
+	k8sVersion, err := GetK8SVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get k8s version, Err: %v", err)
+	}
+	pxSpecURL := reconstructSpecURL(cluster)
+	pxImageList, err := GetImagesFromVersionURL(pxSpecURL, k8sVersion)
+	if err != nil {
+		return fmt.Errorf("failed to get images from version URL: [%v]", err)
+	}
+	pxVersion := GetPortworxVersion(cluster)
+
+	var nodeWiperImage string
+	if opVersion.GreaterThanOrEqual(opVer23_10_2) {
+		if pxVersion.GreaterThanOrEqual(pxVer3_1) {
+			pxImage := cluster.Spec.Image
+			pxVersionStr := strings.Split(pxImage, ":")[len(strings.Split(pxImage, ":"))-1]
+			nodeWiperImage = fmt.Sprintf("portworx/oci-monitor:%s", pxVersionStr)
+		} else {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		}
+	} else if opVersion.LessThan(opVer23_10_2) {
+		if pxVersion.LessThan(pxVer3_1) {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		} else {
+			nodeWiperImage = defaultNodeWiperImage
+		}
+	}
+	logrus.Debugf("Expected nodeWiper image: [%s]", nodeWiperImage)
+	err = validateContainerImageInsidePods(cluster, nodeWiperImage, "px-node-wiper", pods)
+	if err != nil {
+		return fmt.Errorf("failed to validate NodeWiper image: [%v]", err)
+	}
+	logrus.Debugf("NodeWiper images match!")
+
+	return nil
+}
+
+func reconstructSpecURL(cluster *corev1.StorageCluster) string {
+	var pxUrl string
+	pxImage := cluster.Spec.Image
+	pxVersionStr := strings.Split(pxImage, ":")[len(strings.Split(pxImage, ":"))-1]
+	_, err := version.NewSemver(pxVersionStr)
+	if err != nil {
+		logrus.WithError(err).Warnf("Invalid PX version [%s] extracted from image name", pxVersionStr)
+		for _, env := range cluster.Spec.Env {
+			if env.Name == PxImageEnvVarName {
+				pxImage = env.Value
+				pxUrl = fmt.Sprintf("https://edge-install.portworx.com/%s", pxImage)
+			} else if env.Name == PxReleaseManifestURLEnvVarName {
+				pxUrl = strings.TrimSuffix(env.Value, "/version")
```

Comment:
Just noticed, we can't utilize this func https://github.com/libopenstorage/operator/blob/master/pkg/util/test/util.go#L5413 outside of this operator test framework as it takes global `version` param, can we at least use same methods to reconstruct using `net/url`, it will look cleaner

Reply:
For the reconstruction of manifest URL, we just want till the spec endpoint part to be constructed `/versions` & k8s version `?kbver` are appended by the `GetImagesFromVersionURL(pxSpecURL, k8sVersion)` function. The `ConstructVersionURL` you are referring here expects the spenGenURL which we are trying to re-construct here.

As for the air-gapped deployment, I have made the changes to check if cm exists, then read the configmap instead.